### PR TITLE
Update/menu

### PR DESCRIPTION
**Pull request by LuoKevin**
Updated wrappers for the antd menu component

### CODE REVIEW

File: kotlin-antd/src/main/kotlin/antd/menu/MenuItem.kt [hunk 1]
```diff
@@ -18,6 +18,10 @@ external interface MenuItemProps : RProps {
     var onClick: ((param: ClickParam) -> Unit)?
     var onMouseEnter: ((e: ItemClickParam) -> Unit)?
     var onMouseLeave: ((e: ItemClickParam) -> Unit)?
+    var key: Any? /* Object */
+    var itemIcon: Any? /* ReactElement | ((p: MenuItemProps) -> ReactElement */
+    var icon: Any? /* ReactElement */
```
Comment (gad2103) [inline kotlin-antd/src/main/kotlin/antd/menu/MenuItem.kt]: why not make this
`var icon: ReactElement?`
ReactElement is a supported type

File: kotlin-antd/src/main/kotlin/antd/menu/SubMenu.kt [hunk 2]
```diff
@@ -11,12 +11,15 @@ external interface SubMenuProps : RProps {
     var rootPrefixCls: String?
     var className: String?
     var disabled: Boolean?
+    var level: Number?
     var title: Any? /* String | ReactElement */
+    var icon: Any? /* ReactElement */
```
Comment (gad2103) [inline kotlin-antd/src/main/kotlin/antd/menu/SubMenu.kt]: same here.

File: kotlin-antd/antd-samples/src/main/kotlin/samples/menu/StyleDebug.kt [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+}
+
+fun RBuilder.styleDebugChild() = child(StyleDebug::class) {}
+
+fun RBuilder.styleDebug() {
+    styledDiv {
+//        css { +MenuStyles.}
+        styleDebugChild()
+    }
+}
```
Comment (gad2103) [inline kotlin-antd/antd-samples/src/main/kotlin/samples/menu/StyleDebug.kt]: add newline

File: kotlin-antd/antd-samples/src/main/kotlin/samples/menu/StyleDebug.kt [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+            }
+        }
+    }
+}
+
+fun RBuilder.styleDebugChild() = child(StyleDebug::class) {}
+
+fun RBuilder.styleDebug() {
+    styledDiv {
+//        css { +MenuStyles.}
```
Comment (gad2103) [inline kotlin-antd/antd-samples/src/main/kotlin/samples/menu/StyleDebug.kt]: why is this commented?

File: kotlin-antd/src/main/kotlin/antd/menu/MenuItem.kt (hunk 1 above)
Comment (oxiadenine) [reply kotlin-antd/src/main/kotlin/antd/menu/MenuItem.kt]: Yep, it's fine, in the Button component it's like this right now

File: kotlin-antd/src/main/kotlin/antd/menu/SubMenu.kt (hunk 2 above)
Comment (oxiadenine) [reply kotlin-antd/src/main/kotlin/antd/menu/SubMenu.kt]: Same as Menu.kt, it's fine to do `var icon: ReactElement?`

**Review by oxiadenine — APPROVED** [review]

**Review by oxiadenine — CHANGES_REQUESTED** [review]
I checked it in more depth and there is some changes missing:

You have to look from component props that are depending on Rc components and wrap this too, for this component it's necessary to wrap RcMenuItemProps as an additional interface.

You can view how I do it in other components just viewing last commits of updated components.

And as a last minor appreciation, when in the TS definition file (.d.ts), a prop is like this, `icon: React.ReactNode`, commonly in JS it's allowed to write an String or ReactElement type for that variable.

**Pull request closed without merging**